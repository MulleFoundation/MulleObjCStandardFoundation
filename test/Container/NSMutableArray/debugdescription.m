//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>


int main( int argc, const char * argv[])
{
   NSMutableArray   *array;
   NSNumber         *nr;
   NSString         *key;
   NSString         *address;
   NSString         *desc;

   // simple basic test for leakage

   nr    = [NSNumber numberWithInt:1848];
   array = [NSMutableArray arrayWithObject:nr];
   key   = [NSString stringWithUTF8String:"bar"];
   [array addObject:key];

   address = [NSString stringWithFormat:@"%p", array];
   desc    = [array debugDescription];
   desc    = [desc stringByReplacingOccurrencesOfString:address
                                             withString:@"<address>"];
   printf( "%s\n", [desc UTF8String]);

   return( 0);
}
