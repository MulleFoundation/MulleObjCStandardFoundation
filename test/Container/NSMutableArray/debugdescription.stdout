<NSMutableArray <address> "(
    1848,
    bar
)">
